starter-code/lib/harryPotter.js

```diff
@@ -8,3 +8,27 @@
                    "4 October", "Filius Flitwick", "17 October", "Molly Weasley",
                    "30 October", "Bill Weasley", "29 November", "Rubeus Hagrid",
                    "6 December", "Charlie Weasley", "12 December"];
+
+var characterAndBirthday = birthdays.splice(0,2);
+
+console.log(characterAndBirthday);
+
+//
+
+var harryPotter = birthdays.reduce(function(acc,item,index) {
+
+},[]);
+
+///WE THOUGHT THIS COULD BE A GOOD WAY TO GET THERE
+
+var characterAndBirthday1 = birthdays.splice(0,2);
+
+console.log(characterAndBirthday1);
+
+console.log(birthdays);
+
+var characterAndBirthday2 = birthdays.splice(0,2);
+
+console.log(characterAndBirthday2);
+
+console.log(birthdays);
```

Comment:
You almost got it regarding reduce! :)

I will give you my solution using reduce and forEach.

```
// Using reduce:
var matrix = birthdays.reduce(function(acc, item, index, array) {
  if (index % 2 === 0) {
    acc.push([array[index], array[index + 1]]);
  }
  return acc;
}, []);

// Using forEach
var nameBirtPairs = [];
birthdays.forEach(function(element, index, array) {
  if (index % 2 === 0) {
    nameBirtPairs.push([array[index], array[index + 1]]);
  }
});
console.log(nameBirtPairs);

// As we do not want any unnecessary variable in the global scope! We should wrap the forEach method inside a function, like this:
function generateNameBirth(array) {
  var nameBirtPairs = [];
  array.forEach(function(element, index, array) {
    if (index % 2 === 0) {
      nameBirtPairs.push([array[index], array[index + 1]]);
    }
  });
  return nameBirtPairs;
}
console.log(generateNameBirth(birthdays));

```